meson.build

```diff
@@ -18,48 +18,48 @@ if build_machine.system() == 'freebsd'
     add_project_arguments('-DFREEBSD', language : 'c')
 endif
 
-if get_option('with-zstd') == 'disabled'
-    zstd_dep = dependency('', required : false)
+# zstd dependency
+if build_machine.system() == 'windows'
+    zstd_dep = dependency('zstd', modules: 'zstd::libzstd_shared', required : get_option('with-zstd'), fallback: ['zstd', 'libzstd_dep'])
 else
-    if build_machine.system() == 'windows'
-        zstd_dep = dependency('zstd', modules: 'zstd::libzstd_shared', required : get_option('with-zstd') == 'enabled')
-    else
-        zstd_dep = dependency('libzstd', required : get_option('with-zstd') == 'enabled',
-                              default_options: ['bin_programs=false', 'zlib=disabled', 'lzma=disabled', 'lz4=disabled'])
-    endif
-    if zstd_dep.found()
-        add_project_arguments('-DZCHUNK_ZSTD', language : 'c')
-        if zstd_dep.version().version_compare('<=1.4.9')
-            add_project_arguments('-DOLD_ZSTD', language : 'c')
-        endif
+    zstd_dep = dependency('libzstd', required : get_option('with-zstd'), fallback: 'zstd',
+                          default_options: ['bin_programs=false', 'zlib=disabled', 'lzma=disabled', 'lz4=disabled'])
+endif
+if zstd_dep.found()
+    add_project_arguments('-DZCHUNK_ZSTD', language : 'c')
+    if zstd_dep.version().version_compare('<=1.4.9')
+        add_project_arguments('-DOLD_ZSTD', language : 'c')
     endif
 endif
 
+# curl dependency
 if build_machine.system() == 'windows'
-    curl_dep = dependency('curl', modules: 'CURL::libcurl')
+    curl_dep = dependency('curl', modules: 'CURL::libcurl', required : get_option('with-curl'))
 else
-    curl_dep = dependency('libcurl')
+    curl_dep = dependency('libcurl', required : get_option('with-curl'))
+endif
+if not curl_dep.found()
+    curl_dep = disabler()
```

Comment:
Meson 0.49 added the ability to do:

```meson
curl_dep = dependency('libcurl', required: get_option('with-curl'), disabler: true)
```

Then there is no need to check if it is found and overwrite it with a disabler. If it is not found, it will automatically be a disabler.